Makefile

```diff
@@ -1,74 +1,142 @@
-PHP_BIN ?= docker run -it --rm --user $$(id -u):$$(id -g) -v${PWD}:/opt/project -w /opt/project php:8.2-cli php -d memory_limit=1024M
-PHP_PROJECT_BIN = docker run -i --rm --user $$(id -u):$$(id -g) -v${PWD}:/project typo3-docs:local php -d memory_limit=1024M
-## This can be adapted in the .git/hooks/pre-commit call of this step. If you change this line (i.e. new PHP version), also change it in that template
+## Global PHP arguments, applied to both docker and local execution
+PHP_ARGS ?= -d memory_limit=1024M
+
+## Docker wrapper, for raw php commands (so it's not required on the host)
+## This container has no runtime for the `guides` project!
+PHP_BIN ?= docker run -i --rm --user $$(id -u):$$(id -g) -v${PWD}:/opt/project -w /opt/project php:8.1-cli php $(PHP_ARGS)
+
+## Docker wrapper to use for a typo3-docs:local container.
+## This container provides a runtime for the `guides` project
+PHP_PROJECT_BIN ?= docker run -i --rm --user $$(id -u):$$(id -g) -v${PWD}:/project typo3-docs:local
+
+## Docker wrapper to use for a typo3-docs:local container.
+## This container provides a composer-runtime; mounts project on /app
+PHP_COMPOSER_BIN ?= docker run -i --rm --user $$(id -u):$$(id -g) -v${PWD}:/app composer:latest
+
+## These variables can be overriden by other tasks, i.e. by `make PHP_ARGS=-d memory_limit=2G pre-commit-tests`.
+
+## Parse the "make (target) ENV=(local|docker)" argument to set the environment. Defaults to docker.
+ifdef ENV
+	ifeq ($(ENV),local)
+		PHP_BIN = php $(PHP_ARGS)
+		PHP_PROJECT_BIN = php $(PHP_ARGS) ./vendor/bin/guides
+		PHP_COMPOSER_BIN = composer
+		ENV_INFO=ENVIRONMENT: Local (also DDEV)
+	else
+		ENV_INFO=ENVIRONMENT: Docker
+	endif
+else
+	ENV_INFO=ENVIRONMENT: Docker (default)
+endif
 
 .PHONY: help
 help: ## Displays this list of targets with descriptions
+	@echo "You prepend/append the argument 'ENV=(local|docker)' to each target. This specifies,"
+	@echo "whether to execute the target within your local environment, or docker (default).\n"
 	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[32m%-30s\033[0m %s\n", $$1, $$2}'
 
+## LIST: Targets that can be executed directly
+
+.PHONY: show-env
+show-env: ## Shows PHP environment options (buildinfo)
+	@echo "$(ENV_INFO)"
+	@echo "Base PHP:"
+	$(PHP_BIN) --version
+	@echo ""
+
+	@echo "Project within Docker:"
+    # TODO: See Issue #72, does not work yet
+	docker run --rm --user $$(id -u):$$(id -g) -v${PWD}:/project typo3-docs:local --version
```

Comment:
`--user` must be omitted, as the docker image will detect the correct user.

Reply:
Hm, if I change that to:

`docker run --rm -v${PWD}:/project typo3-docs:local --version`

it only reports:

```
ould not detect the owner of /project. Did you mount your project?.
Run this container with "docker run --rm  --volume ${PWD}:/project ghcr.io/typo3-documentation/render-guides:main"
```

(note that I ran the above with "typo3-docs:local" - and did you check out Issue #72 - this has some more background?)